SpatialGDK/Source/SpatialGDKFunctionalTests/Private/SpatialFunctionalTest.cpp

```diff
@@ -367,6 +369,31 @@ void ASpatialFunctionalTest::FinishTest(EFunctionalTestResult TestResult, const
 	}
 }
 
+void ASpatialFunctionalTest::AddExpectedError(const FString& ExpectedErrorPattern, int NumOccurences /*= 1*/,
+											  bool bAllowPartialMatch /*= true*/)
+{
+	if (HasAuthority())
+	{
+		ExpectedErrors.Add({ ExpectedErrorPattern, NumOccurences, bAllowPartialMatch });
+	}
+}
+
+void ASpatialFunctionalTest::SetupExpectedErrors()
+{
+	if (HasAuthority())
+	{
+		FAutomationTestBase* CurrentTest = FAutomationTestFramework::Get().GetCurrentTest();
+		check(CurrentTest != nullptr);
+		for (const auto& ExpectedError : ExpectedErrors)
+		{
+			CurrentTest->AddExpectedError(
+				ExpectedError.ErrorPattern,
+				ExpectedError.bAllowPartialMatch ? EAutomationExpectedErrorFlags::Contains : EAutomationExpectedErrorFlags::Exact,
```

Comment:
I think we could expose `EAutomationExpectedErrorFlags` directly, but minor thing.

Reply:
I didn't specifically to not change the engine, if you prefer I can, but need to change it to enum class, etc.

Reply:
Hm? Why does this require changing the engine? I thought this was exposed for use in the AutomationTests. Maybe will need to catch up "offline", but I think there should exist a configuration that allows us to use it :D.